controllers/hotel_controller.go

```diff
@@ -0,0 +1,85 @@
+package controllers
+
+import (
+	cError "github.com/couchbase-examples/golang-quickstart/errors"
+	"github.com/couchbase-examples/golang-quickstart/models"
+	services "github.com/couchbase-examples/golang-quickstart/service"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+type HotelController struct {
+	HotelService services.IHotelService
+}
+
+func NewHotelController(airlineService services.IHotelService) *HotelController {
+	return &HotelController{
+		HotelService: airlineService,
+	}
+}
+
+// SearchByName
+//
+// @Summary Search by hotel name
+// @Description Search for hotels based on their name.
+// @Tags Hotel
+// @Produce json
+// @Param name query string true "name search"
+// @Success 200 {object} []string
+// @Failure 400 "Bad Request"
+// @Failure 500 "Internal Server Error"
+// @Router /api/v1/hotel/autocomplete [get]
+func (h *HotelController) SearchByName() gin.HandlerFunc {
+	return func(context *gin.Context) {
+		name := context.Query("name")
+		if name == "" {
+			context.JSON(http.StatusBadRequest, cError.Errors{Error: "name query parameter is required"})
+			return
+		}
+		hotels, err := h.HotelService.SearchByName(name)
+		if err != nil {
+			context.JSON(http.StatusInternalServerError, cError.Errors{
+				Error: err.Error(),
+			})
+			return
+		}
+		context.JSON(http.StatusOK, hotels)
+	}
+}
+
+// Filter controller
+//
+// @Summary Fetch hotels with multiple filters
+// @Description Fetch hotels using various filters such as name, title, description, country, state and city.
```

Comment:
Same as above, Can you please add more description like other routes. For eg: Code and method

Reply:
done